dap_aria_mapping/analysis/app/pages/2_Change_Makers.py

```diff
@@ -18,6 +24,121 @@
     page_icon=icon
 )
 
+@st.cache_data(show_spinner = "Loading Data")
+def load_networks(dataset: str) -> nx.Graph:
+    """get the collaboration network for either patents or publications
+
+    Args:
+        dataset (str): "Academia" for publication network or "Industry" for patent network 
+
+    Returns:
+        nx.Graph: collaboration network
+    """
+    return get_collaboration_network(dataset)
+
+@st.cache_data(show_spinner = "Loading Data")
+def load_filter_data(dataset: str) -> Tuple[pl.DataFrame, List[str]]:
+    """gets lookup table of domains to areas and topic names to populate filters, 
+    as well as initial list of all domains for starting filter value
+
+    Args:
+        dataset: "Industry" or "Academia" specified by filter
+    Returns:
+        Tuple[pl.DataFrame, List[str]]: lookup table for filtering and unique domains
+    """
+    if dataset == "Industry":
+        data = get_topic_domain_area_lookup("patents")
+    elif dataset == "Academia":
+        data = get_topic_domain_area_lookup("publications")
+    unique_domains = data["domain_name"].unique().to_list()
+    return data, unique_domains
+
+@st.cache_data(show_spinner = "Filtering Data to Domain")
+def filter_by_domain(domain: str, dataset: str) -> Tuple[nx.Graph, List[str]]:
```

Comment:
Why not import the `filter_data` object that you create in the stream below, instead of having to re-run it whenever the domain changes?

Reply:
I think I had to do that to make it so you could go back to a new domain after you had already selected one